### DISCUSSION TITLE
MLinPL 2023 - feedback and ideas!

### DISCUSSION THREAD

**Discussion by Calychas** [kraina-ai/srai#389] [Ideas]
Hello everyone! 
We would like to warmly invite you to share feedback from the tutorial and some ideas for SRAI usage: 

* Do you have any feedback regarding the tutorial?
* What are your use cases?
* Could SRAI help you solve your problems?
* Do you have any tips?

We would love to hear about them to include them in future SRAI development. 

Thanks, everyone, for the contribution. It is greatly appreciated!

**Comment by JakubCha** [comment]
Hello guys!

> Do you have any feedback regarding the tutorial?

It was intensive. I learned a lot regarding the possibilities that `srai` gives to present geospatial engineers so it was time well spent!

> What are your use cases?

I tend to use more imagery (satellite, aerial) data so it would be nice to see them included in your library, connected with present features for vector data.

> Could SRAI help you solve your problems?

Definitely. I will try to use `srai` in an ML pipeline.

> Do you have any tips?

If I had to change anything regarding tutorial I would focus more on srai functionalities, less on (geo)pandas hustle and bustle. At the same time, I know that your focus was also to "introduce geospatial in ML" as a domain so it's hard to balance and make everything in a limited timeframe.

Thank you!

**Comment by pmallas** [comment]
Howdy SRAI folks!

Thanks for a great library.

Do you have any feedback regarding the tutorial?
Well, it was great.  Very in depth and covered a lot of material. 

In the answers of the bike example, I believe there was a typo:
```
# Mark remaining regions as negative
negative_samples = regions_gdf.copy()[~regions_gdf.index.isin(positive_samples_new.index)]
```

There is an "_new" appended to the DataFrame name.  

That of course is minor.  But the next cell fails for me and appears to be a bug:

```
embeddings = embedder.transform(
    regions_gdf=regions_gdf, features_gdf=features_gdf, joint_gdf=joined_features
 )

....

File [~\Miniconda3\envs\ais\lib\site-packages\srai\neighbourhoods\_base.py:154](http://localhost:8891/lab/tree/~/Miniconda3/envs/ais/lib/site-packages/srai/neighbourhoods/_base.py#line=153), in Neighbourhood._handle_center(self, index, distance, neighbours, at_distance, include_center_override)
    152 else:
    153     if at_distance:
--> 154         neighbours.discard(index)
    155     else:
    156         if include_center:

AttributeError: 'list' object has no attribute 'discard'
```

Thanks
Paul

**Reply by gegen07** [reply]
See the issue #431!

**Reply by pmallas** [reply]
Ahh, thanks.  I should have looked harder.

**Reply by Calychas** [reply]
Heyo! This will be resolved in 0.7.1 - https://github.com/kraina-ai/srai/pull/437
